Content.Shared/Maps/TileSystem.cs

```diff
@@ -20,6 +22,17 @@ public sealed class TileSystem : EntitySystem
     [Dependency] private readonly SharedDecalSystem _decal = default!;
     [Dependency] private readonly SharedMapSystem _maps = default!;
     [Dependency] private readonly TurfSystem _turf = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+    private int _tileStackingHistorySize;
+
+    //Tile history for deconstruction.
+    private readonly Dictionary<(EntityUid Grid, Vector2i Indices), Stack<string>> _tileBaseTurfHistory = new();
```

Comment:
in general you should not be storing game data on systems like this.

Instead, you should be creating a new component that holds this data for a single grid and then adding it to newly-made grid entities. Then, retrieve the component and access the data instead of just reading it off the system.

That will also enable you to properly network the data to the clients, as this system will currently heavily mispredict in an actual round.